server/task.go

```diff
@@ -297,7 +298,20 @@ func (s *Server) registerTaskRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to update task status").SetInternal(err)
 		}
 		if task == nil {
-			return echo.NewHTTPError(http.StatusUnauthorized, fmt.Sprintf("Task ID not found: %d", taskID))
+			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Task ID not found: %d", taskID))
+		}
+
+		issue, err := s.IssueService.FindIssue(ctx, &api.IssueFind{
+			PipelineID: &task.PipelineID,
+		})
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to find issue").SetInternal(err)
+		}
+		if issue == nil {
+			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Issue not found by pipeline ID: %d", task.PipelineID))
+		}
+		if issue.AssigneeID != currentPrincipalID {
```

Comment:
https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/frontend/src/components/Issue/IssueStatusTransitionButtonGroup.vue?L217

The logic is more complex.

The assignee can be a system bot if the issue is triggered by VCS, and if we configure the changing environment requires review, then DBA or owner is also allowed to change the task status.

Reply:
When the assigneeID is bot and the current user is DBA or owner, changing the task status is allowed. Otherwise, we only allow assignee changes. wdyt?

(BTW, I find that there is no "system account" in UI. https://github.com/bytebase/bytebase/blob/main/store/migration/10001__init_schema.sql#L711)

Reply:
Maybe we can update the comment there (in a different change). Someone (DBA/owner) should pass an issue another non-robot DBA/owner.